samza-core/src/test/java/org/apache/samza/clustermanager/TestContainerPlacementActions.java

```diff
@@ -266,9 +274,10 @@ public Void answer(InvocationOnMock invocation) {
   @Test(timeout = 30000)
   public void testActionQueuingForConsecutivePlacementActions() throws Exception {
     // Spawn a Request Allocator Thread
-    Thread requestAllocatorThread = new Thread(
-        new ContainerPlacementRequestAllocator(containerPlacementMetadataStore, cpm, new ApplicationConfig(config)),
-        "ContainerPlacement Request Allocator Thread");
+    ContainerPlacementRequestAllocator requestAllocator =
+        new ContainerPlacementRequestAllocator(containerPlacementMetadataStore, cpm, new ApplicationConfig(config));
+    Thread requestAllocatorThread = new Thread(requestAllocator, "ContainerPlacement Request Allocator Thread");
+
```

Comment:
Can this be extracted at the class level and initialize before every tests?

Reply:
No this is not used by every tests only two tests use this route of injecting control messages not all the tests